File: noronha_expression_evaluation.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    int a = 5, b = 10, c = 9, d = 6, e = 4;

    int exprA = ((a * b) - 1) + c;
    printf("Expression a: %d\n", exprA);
    
    int exprB = ((a * (b - 1)) / c) % d;
    printf("Expression b: %d\n", exprB);
    
    int exprC = ((a - b) / c) & ((d * e / a) - 3);
    printf("Expression c: %d\n", exprC);
    
    bool exprD = (-a != 0) || ((c == d) && (e != 0));
    printf("Expression d: %d\n", exprD);
    
    bool exprE = ((a > b) ^ (c != 0)) || (d <= 17);
    printf("Expression e: %d\n", exprE);
    
    int exprF = (-a) + b;
    printf("Expression f: %d\n", exprF);
    
    return 0;
}

File: noronha_expression_evaluation.java
public class noronha_expression_evaluation {
    public static void main(String[] args) {
        int a = 5, b = 10, c = 9, d = 6, e = 4;
        
        int exprA = ((a * b) - 1) + c;
        System.out.println("Expression a: " + exprA);
        
        int exprB = ((a * (b - 1)) / c) % d;
        System.out.println("Expression b: " + exprB);
        
        int exprC = ((a - b) / c) & ((d * e / a) - 3);
        System.out.println("Expression c: " + exprC);

        boolean exprD = (-a != 0) || ((c == d) && (e != 0));
        System.out.println("Expression d: " + exprD);
        
        boolean exprE = ((a > b) ^ (c != 0)) || (d <= 17);
        System.out.println("Expression e: " + exprE);
        
        int exprF = (-a) + b;
        System.out.println("Expression f: " + exprF);
    }
}
